network-core/src/lib.rs

```diff
@@ -5,11 +5,28 @@
 #[macro_use]
 extern crate prost_derive;
 
+use chain_core::property::{Deserialize, Serialize};
+
 pub mod error;
 
 pub mod client;
 pub mod server;
 
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct NodeId(u128);
+
+impl NodeId {
+    pub fn from_bytes(high: u64, low: u64) -> Self {
```

Comment:
These are not bytes. There may be a `fn from_bytes(bytes: &[u8; 16])`, or more conveniently `fn from_bytes(bytes: &[u8])` with a size assertion.

I'd use the name `from_halves`.